### ISSUE TITLE
Prometheus annotation support

### ISSUE THREAD

**Issue by a-thaler** [kyma-project/kyma#11300]
**Description**
The prometheus annotations are quite popular nowadays and are an alternative to the ServiceMonitor definition style. So instead of defining a concrete scrape configuration via a ServiceMonitor, there is a scrape config based on the prometheus service discovery which dynamically selects the endpoints to scrape.

That concept is interesting in regards to providing the feature out-of-the-box for kyma application developers. Furthermore it might be an interesting substitute in case the prometheus-operator is not a mandatory element of the kyma stack anymore (prometheus might be replaced with the openTelemetryCollector)

Goal: Figure out if additional support of prometheus metrics annotationsis feasible and valuable. You can deploy an application with annotations and metrics are getting scraped. Furthermore see if the serviceMonitors for the system components can be replaced by them so that the base kyma setup is not based on ServiceMonitors at all.

For that:
- Understand the service discovery mechanism of prometheus and come up with a scrqape config supporting annotations
- Check istio mergeMetrics option, decide if it must be switched on.. can istio scraping be separated from application metric scraping? The istio-prometheus might scrape the istio ports directly while the main prometheus is based on annotations? 
- Check for all podmonitors we have, can they be substituted
- What is with advanced ServiceMonitor options like metricRelabelling? Can we live without it, what is the replacement?
- Do we have components with annotations in place which might collide with existing scrape approach?
- Check prometheus stability

Major question to answer: Should we support annotations already additionally? Can it be a full replacement long-term thinking of not having the prometheus-operator in the cluster?

**Reasons**
Prometheus-operator might not be used in future, popular charts comes with annotations only

**Attachments**

**Comment by a-thaler** [comment]
The feature should be the main scenario for the future metrics support as described in https://github.com/kyma-project/kyma/issues/13079.
However, we should not add it as a feature to the current stack as it will increase the maintenance effort much more. Instead we should start learning with the approach and already start decoupling the system components from the monitoring chart, so applying the feature for the internal components only.

I created a PR which is doing exactly that. It introduced a new scrape job not based on serviceMonitors which discovers service endpoints by annotation. The job discovers only the system namespaces. Then I replaced the servicemonitors by annotations. See https://github.com/kyma-project/kyma/pull/16247

The only downside I found is that it is not possible to keep the container and endpoint label on the metric, this is not relevant for the existing metrics.

**Comment by skhalash** [comment]
I split https://github.com/kyma-project/kyma/pull/16247 into fine-grained component-based PRs. Otherwise, it was impossible to merge it. What's left is the monitoring component, which has to be handled separately.

**Comment by a-thaler** [comment]
Re-stating my comment in-between to sum up the story:

The annotation support should be a main scenario for the future metrics support as described in https://github.com/kyma-project/kyma/issues/13079.
However, we should not add it as a feature to the current stack as it will increase the maintenance effort much more. Instead we should start learning with the approach and already start decoupling the system components from the monitoring chart, so applying the feature for the internal components only.

In that story, we introduced a new scrape job not based on serviceMonitors which discovers service endpoints by annotation. The job discovers only the system namespaces. Then all servicemonitors of kyma components got replaced by annotations. With that, kyma modules are freed up by prometheus CRDs and decoupled from the monitoring chart, which is essential for the modularization epic.

The only downside is that it is not possible to keep the container and endpoint label on the metric, however, this is not relevant for the existing metrics.
